src/main/java/org/venice/beachfront/bfapi/BfApiConfig.java

```diff
@@ -92,4 +98,183 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
+
+			final List<HttpMessageConverter<?>> messageConverters = new ArrayList<HttpMessageConverter<?>>();
+			messageConverters.add(new StringHttpMessageConverter());
+			messageConverters.add(new MappingJackson2HttpMessageConverter());
+			restTemplate.setMessageConverters(messageConverters);
+
+			return restTemplate;
+		}
+
+		@Bean
+		public AuthHeaders authHeaders() {
+			return new PKIAuthHeaders();
+		}
+
+		protected KeyStore getStore()
+				throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
+			final KeyStore store = KeyStore.getInstance(KeyStore.getDefaultType());
+			final InputStream inputStream = getClass().getClassLoader().getResourceAsStream(keystoreFileName);
+			try {
+				store.load(inputStream, keystorePassphrase.toCharArray());
+			} 
+			finally {
+				inputStream.close();
+			}
+
+			return store;
+		}
+
+		protected class MyCookieSpec extends DefaultCookieSpec {
+
+			@Override
+			public void validate(Cookie c, CookieOrigin co) throws MalformedCookieException {
+				// Do nothing; accept all cookies
+			}
+		}
+
+		protected class MySpecProvider implements CookieSpecProvider {
+
+			@Override
+			public CookieSpec create(HttpContext context) {
+
+				return new MyCookieSpec();
+			}
+		}
+
+		protected class MyRedirectStrategy extends DefaultRedirectStrategy {
+
+			@Override
+			protected boolean isRedirectable(final String method) {
+
+				return true;
+			}
+
+			@Override
+			public HttpUriRequest getRedirect(final HttpRequest request, final HttpResponse response,
+					final HttpContext context) throws ProtocolException {
+
+				final URI uri = getLocationURI(request, response, context);
+				final String method = request.getRequestLine().getMethod();
+
+				if (method.equalsIgnoreCase(HttpHead.METHOD_NAME)) {
+					return new HttpHead(uri);
+				} 
+				else if (method.equalsIgnoreCase(HttpGet.METHOD_NAME)) {
+					return new HttpGet(uri);
+				}
+				else {
+					final int status = response.getStatusLine().getStatusCode();
+
+					if (status == HttpStatus.SC_TEMPORARY_REDIRECT || status == HttpStatus.SC_MOVED_PERMANENTLY
+							|| status == HttpStatus.SC_MOVED_TEMPORARILY) {
+						return RequestBuilder.copy(request).setUri(uri).build();
+					} 
+					else {
+						return new HttpGet(uri);
+					}
+				}
+			}
+		}
+	}
+
+	private static ConnectionKeepAliveStrategy getKeepAliveStrategy() {
+		return (HttpResponse response, HttpContext context) -> {
+			HeaderElementIterator it = new BasicHeaderElementIterator(response.headerIterator(HTTP.CONN_KEEP_ALIVE));
+			while (it.hasNext()) {
+				final HeaderElement headerElement = it.nextElement();
+				final String param = headerElement.getName();
+				final String value = headerElement.getValue();
+				if (value != null && param.equalsIgnoreCase("timeout")) {
+					return Long.parseLong(value) * 1000;
+				}
+			}
+			return 5 * 1000; // TODO: Probably want this configurable
```

Comment:
Yeah, probably want to move the default timeout to the properties file

Reply:
This was leftover from the original code, I'll put it in a properties file.